### PR TITLE
Rework map data load/save.

### PR DESCRIPTION
**Pull request by pchote**
This reorganizes the map fields into a semblance of order, and, where possible, makes them readonly or `{ private set; }`.  Also tidies up the TS map importer a bit.

### CODE REVIEW

File: OpenRA.Game/Map/Map.cs [hunk 1]
```diff
@@ -66,28 +67,185 @@ public enum MapVisibility
...
+			new MapField("MapFormat"),
+			new MapField("RequiresMod"),
+			new MapField("Title"),
+			new MapField("Author"),
+			new MapField("Tileset"),
+			new MapField("MapSize"),
+			new MapField("Bounds"),
+			new MapField("Visibility"),
+			new MapField("Type"),
+			new MapField("LockPreview", required: false, ignoreIfValue: "False"),
```
Comment (Phrohdoh) [inline OpenRA.Game/Map/Map.cs]: Does this subtly enforce case with `True`/`False`?

Comment (pchote) [inline OpenRA.Game/Map/Map.cs]: The yaml saver already does that.

**Comment by Phrohdoh** [comment]
:+1:

**Comment by obrakmann** [comment]
Looks fine here as well, :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/7b6845c062dce19a38d2f955bccb0d12badfd1f2)

**Pull request merged**